Source/Core/DolphinQt/Config/ARCodeWidget.cpp

```diff
@@ -230,6 +231,11 @@ void ARCodeWidget::LoadCodes()
     const Common::IniFile game_ini_default =
         SConfig::LoadDefaultGameIni(m_game_id, m_game_revision);
     m_ar_codes = ActionReplay::LoadCodes(game_ini_default, game_ini_local);
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+    if (!m_restart_required)
+      AchievementManager::GetInstance().FilterApprovedARCodes(m_ar_codes, m_game_id);
+#endif  // USE_RETRO_ACHIEVEMENTS
```

Comment:
You're still filtering here and in GeckoCodeWidget. Have you really tested that this works as intended, without the INI automatically changing?